python/cudf/cudf/utils/metadata/orc_column_statistics.proto

```diff
@@ -0,0 +1,61 @@
+message IntegerStatistics  {
```

Comment:
I'm not sure we want to add the complexity of protobuf into here as it would make building from source just that much harder. Is protobuf needed here?

Reply:
In short - yes, ORC metadata is stored in protobuf format. I had briefly outlined the alternatives above-
> - Using PyORC which is (1) a small open-source library and (2) a binding to ORC Core C++ with same limitations
> - Using ORC Core C++ which is (1) an additional external dependency and (2) does not expose a way to read stripe-level statistics without also reading statistics for all row groups within given stripes (This resulted in a ~20% overhead for #6116 which was reduced to ~0.1% with this PR)
> - Implementing parsing of ColumnStatistics directly in libcudf instead of using a generated pure Python ProtoBuf parser may be tricky with given refactor while this is simpler (only 2 affected files in libcudf) and allows us to move ahead with #6116 for now

Also, protobuf is not being included as a dependency - the build process should be same. Rather, `protoc` was used to generate a pure Python module for parsing `orc_column_statistics`. In this PR, we are checking in this generated module which was generated using the command in the PR comment at top. @quasiben had mentioned to me that he's seen this done elsewhere (his earlier comment - https://github.com/rapidsai/cudf/pull/6142#issuecomment-687307145).

I understand this probably isn't ideal (ideally, the parsing should happen directly in libcudf using the existing small protobuf parser) but given the ongoing cuIO refactoring, do you think this is okay for now and implementing `ColumnStatistics` parsing in libcudf can happen in a later PR?

Reply:
Thanks for the great explanation. Will take another pass at reviewing now 😄